packages/passport-interface/src/PCDIntegration.ts

```diff
@@ -33,13 +34,48 @@ export function usePassportPCD() {
     function receiveMessage(ev: MessageEvent<any>) {
       // This next line is important. Extensions including Metamask apparently
       // send messages to every page. Ignore those.
-      if (!ev.data.encodedPcd) return;
+      if (!ev.data.encodedPCD) return;
       console.log("Received message", ev.data);
-      setPcdStr(ev.data.encodedPcd);
+      setPcdStr(ev.data.encodedPCD);
     }
     window.addEventListener("message", receiveMessage, false);
     return () => window.removeEventListener("message", receiveMessage);
   }, []);
 
   return pcdStr;
 }
+
+/**
+ * React hook that listens for PCDs and/or PendingStampPCDs returned by the passport
+ * to the application. The former is returned if client-side proofs are selected, the
+ * latter if server-side proofs are selected.
+ */
+export function usePassportPCDOrStamp(): [
```

Comment:
I think this should probably be a modification of `useProof` vs. a new hook. I think it should also return an object rather than an array, so we can name these returned variables. Finally I don't think I understand why you need to return `setPcdStr`. I see that you're using it later, but I think that the place which should handle loading pending PCDs is probably also this hook, as opposed to a functional component that loads the pending PCD and then sets it in this hook. It's quite confusing to understand what's going on otherwise.